packages/expo-audio/src/index.ts

```diff
@@ -15,22 +16,30 @@ import { createRecordingOptions } from './utils/options';
 import { resolveSource } from './utils/resolveSource';
 
 export function useAudioPlayer(
-  source: AudioSource | string | number | null = null,
-  statusListener?: (status: AudioStatus) => void
-): AudioPlayer {
+  source: AudioSource | string | number | null = null
+): readonly [AudioPlayer, AudioStatus] {
   const parsedSource = resolveSource(source);
   const player = useReleasingSharedObject(() => {
     return new AudioModule.AudioPlayer(parsedSource);
   }, [JSON.stringify(parsedSource)]);
 
-  useEffect(() => {
-    const subscription = player.addListener('onPlaybackStatusUpdate', (status) => {
-      statusListener?.(status);
-    });
-    return () => subscription.remove();
-  }, [player.id]);
+  const status = useEvent(player, 'onPlaybackStatusUpdate', {
+    id: player.id,
+    currentTime: player.currentTime,
+    status: 'unknown',
+    timeControlStatus: 'unknown',
+    reasonForWaitingToPlay: 'unknown',
+    mute: player.muted,
+    duration: player.duration,
+    playing: player.playing,
+    loop: player.loop,
+    isBuffering: player.isBuffering,
+    isLoaded: player.isLoaded,
+    playbackRate: player.playbackRate,
+    shouldCorrectPitch: player.shouldCorrectPitch,
+  });
 
-  return player;
+  return [player, status] as const;
```

Comment:
Are you sure you want to return an array here? As a user, I would rather expect only the player. We could have another `useAudioPlayerStatus(player)` if needed.

Reply:
I would have thought the player and it's status are tightly coupled but we can do it like that. You'll likely never use the player in isolation so you'll need both each time

Reply:
I don't think it's too unexpected, `useState`, `useReducer`, `useTransition` etc. all do this